lib/view.cc

```diff
@@ -185,9 +191,10 @@ View View::getMessages(const std::vector<std::string> &topics) {
 
       for (const auto &connection_record : bag->topic_connection_map_.at(topic)) {
         for (const auto &block : connection_record->blocks) {
-          bag_wrappers_[bag]->chunks_to_parse.emplace(block.into_chunk);
+          if (block.into_chunk->info.end_time > start_time && block.into_chunk->info.start_time < end_time) {
```

Comment:
I'd recommend doing >= and <= to be inclusive.

Reply:
I also think you might also want further filtering when yielding the actual messages within the chunk, and not just filtering at the chunk level.

Reply:
Finally, this filter will remove any chunks that straddle the edge of the filter - I don't think thats the behaviour we want. We should be reading any chunks that contain any data within the start and end time.

Reply:
Ah we can filter when we yield, that's a good idea.

Reply:
So save the start/end timestamps to the view, and when yielding do another bounds check?

Reply:
Ya exactly - I think that would be the right approach in this case! We definitely still want to filter out chunks for some performance gains, but then a secondary filter will do the final checks for us.